kvbc/include/ClientImp.h

```diff
@@ -40,7 +40,9 @@ class ClientImp : public IClient {
 
  protected:
   ClientImp() = default;
-  ~ClientImp() override = default;
+  ~ClientImp() override {
+    if (comm_) delete comm_;
```

Comment:
Nit: no need to verify if comm_ is not null.

Reply:
it's double checking to prevent double free if someone would change something in a different location. the comm obj is passed through various factories so I thought it would be safest to add the if.

Reply:
1. The 'if' will not prevent the double-free in case comm_ is not set to nullptr after deletion. 
2. In case comm_ equals nullptr delete ignores it.

Reply:
I see... so there's no real way to verify the obj wasn't freed without smart ptrs, right?